frontend/components/tree/js/tree.service.ts

```diff
@@ -994,70 +1100,256 @@ export class TreeService {
-	 * @param node	Node to toggle.
+	 * Select a series of nodes by an array of shared IDs (rather than unique IDs)
+	 * @param objects	Nodes to select
+	 * @param multi	Is multi select enabled
+	 * @param additive whether the selection should reset or keep add selections
+	 * @param colour the colour to highlight
 	 */
-	public updateModelStateShown(node) {
-		if (node.toggleState === "visible") {
-			this.clickedShown[node._id] = node;
-		} else {
-			delete this.clickedShown[node._id];
+	public selectNodesBySharedIds(
+		objects: any[], multi: boolean, additive: boolean,
+		colour: number[], forceReHighlight: boolean,
+	) {
+
+		return this.getNodesFromSharedIds(objects)
+			.then((nodes) => {
+				return this.selectNodes(nodes, multi, additive, colour, forceReHighlight);
+			})
+			.catch((error) => {
+				console.error(error);
+			});
+	}
+
+	/**
+	 * Highlight a series of nodes based on shared IDs (rather than unique IDs)
+	 * @param objects	Nodes to select
+	 * @param multi	Is multi select enabled
+	 * @param colour the colour to highlight
+	 * @param forceReHighlight force a rehighlighting to a new colour (overides toggle)
+	 */
+	public highlightNodesBySharedId(
+		objects: any[], multi: boolean, colour: number[], forceReHighlight: boolean,
+	) {
+		return this.getNodesFromSharedIds(objects)
+			.then((nodes) => {
+				this.highlightNodes(nodes, multi, colour, forceReHighlight);
+			})
+			.catch((error) => {
+				console.error(error);
+			});
+
+	}
+
+	/**
+	 * Isolate selected objects by their shared IDs
+	 */
+	public isolateNodesBySharedId(objects) {
+
+		return this.getNodesFromSharedIds(objects)
+			.then((nodes) => {
+
+				// Hide all
+				this.hideAllTreeNodes(false); // We can just reset the state without hiding in the UI
+				// Show selected
+
+				this.setCurrentSelectedNodes(nodes);
+				this.showTreeNodes(nodes);
+
+			})
+			.catch((error) => {
+				console.error(error);
+			});
+
+	}
+
+	/**
+	 * Hide series of nodes by an array of shared IDs (rather than unique IDs)
+	 * @param objects objects to hide
+	 */
+	public hideBySharedId(objects: any[]) {
+
+		return this.getNodesFromSharedIds(objects)
+			.then((nodes) => {
+
+				this.hideTreeNodes(nodes);
+
+			})
+			.catch((error) => {
+				console.error(error);
+			});
+
+	}
+
+	/**
+	 * Show a series of nodes by an array of shared IDs (rather than unique IDs)
+	 * @param objects objects to show
+	 */
+	public showBySharedId(objects: any[]) {
+
+		if (!objects || objects.length === 0) {
+			return;
 		}
+
+		return this.getNodesFromSharedIds(objects)
+			.then((nodes) => {
+
+				this.hideAllTreeNodes(false);
+				this.showTreeNodes(nodes);
+
+			})
+			.catch((error) => {
+				console.error(error);
+			});
+
+	}
+
+	/**
+	 * Highlight a series of nodes by an array of shared IDs (rather than unique IDs)
+	 * @param objects objects to show
+	 */
+	public highlightsBySharedId(objects: any) {
+
+		// TODO: This could probably be refactored
```

Comment:
this really should be rewritten to utilise the sharedID to node function and call select nodes with an array instead of calling select node separately!

Reply:
Yeah definitely. Also probably only need to call expand to selection on the last node